src/crimson/os/seastore/onode_manager/staged-fltree/fltree_onode_manager.h

```diff
@@ -78,27 +80,142 @@ struct FLTreeOnode final : Onode, Value {
     return *read_payload<onode_layout_t>();
   }
 
-  onode_layout_t &get_mutable_layout(Transaction &t) final {
+  template <typename layout_func_t, typename recorder_func_t>
+  void with_mutable_layout(
+    Transaction &t,
+    layout_func_t &&layout_func,
+    recorder_func_t &&recorder_func) {
     assert(status != status_t::DELETED);
     auto p = prepare_mutate_payload<
       onode_layout_t,
       Recorder>(t);
     status = status_t::MUTATED;
-    return *reinterpret_cast<onode_layout_t*>(p.first.get_write());
-  };
-
-  void populate_recorder(Transaction &t) {
-    assert(status == status_t::MUTATED);
-    auto p = prepare_mutate_payload<
-      onode_layout_t,
-      Recorder>(t);
+    layout_func(*reinterpret_cast<onode_layout_t*>(p.first.get_write()));
     if (p.second) {
-      p.second->record_delta(
-        p.first);
+      recorder_func(p.first, *p.second);
     }
     status = status_t::STABLE;
```

Comment:
nit, see https://github.com/ceph/ceph/pull/53234#discussion_r1349896192, looks STABLE and MUTATED doesn't need to be distinguished.

Reply:
Done.